### PR TITLE
GitHub Actions: Installing frozenlist fails on python3.11-beta.5

### PR DESCRIPTION
**Pull request by cclauss**
An on-demand GitHub Action...
```yaml
name: python_3_11-dev
on:
  workflow_dispatch:
jobs:
  python_3_11-dev:
    strategy:
      fail-fast: false
      matrix:
        package:
          - cchardet  # Passes if cython>=0.29.30
          - frozenlist  # Fails
          - yarl  # Fails
          - git+https://github.com/PyYoshi/cChardet.git  # Passes!!!
          - git+https://github.com/aio-libs/frozenlist.git  # Fails
          - git+https://github.com/aio-libs/yarl.git  # Fails
    runs-on: ubuntu-latest
    steps:
      - uses: actions/setup-python@v4
        with:
          python-version: 3.11-dev
      - run: python3.11 -m pip install --upgrade pip wheel
      - run: python3.11 -m pip install --upgrade Cython>=0.29.30
      - run: python3.11 -m pip install ${{ matrix.package }}
```
Test a workaround for #6600
* https://github.com/aio-libs/aiohttp/pull/6830/files
* https://github.com/aio-libs/aiohttp/runs/7379083164?check_suite_focus=true

https://github.com/aio-libs/yarl/issues/706#issuecomment-1187623862

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -168,9 +168,15 @@ jobs:
     - name: Update pip, wheel, setuptools, build, twine
       run: |
         python -m pip install -U pip wheel setuptools build twine
+    - name: Test a workaround for #6600
+      if: startsWith(matrix.pyver, '3.11')
+      run: python -m pip install --upgrade cchardet
```
Comment (cclauss) [inline .github/workflows/ci.yml]: This fails on Python 3.11-beta.4

Comment (Dreamsorcerer) [reply .github/workflows/ci.yml]: See #6819.
Feel free to add something to remove it from installation in 3.11 for now.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -168,6 +172,9 @@ jobs:
     - name: Update pip, wheel, setuptools, build, twine
       run: |
         python -m pip install -U pip wheel setuptools build twine
+    - name: Test a workaround for #6600
+      if: startsWith(matrix.pyver, '3.11')
+      run: python -m pip install cython
```
Comment (webknjaz) [inline .github/workflows/ci.yml]: This doesn't seem necessary. Cythonize is called below depending on the `no-extensions` matrix factor. `cython` itself does get installed without an explicit separate command.
Instead, it should be updated in `requirements/cython.txt` and `requirements/constraints.txt`.

Comment (cclauss) [inline .github/workflows/ci.yml]:
```suggestion
```

**Comment by Dreamsorcerer** [comment]
You might want to revert the mypy version change, I'll fix the new errors in a separate PR.

**Comment by owocado** [comment]
I think the original post might need to be updated to mention that pip install of latest version of `yarl` (>=1.8.1) passes with success on latest python v3.11-dev (3.11.0.beta5) since the OP seems outdated due to `# Fails` comments on 2 yarl entries.

**Comment by cclauss** [comment]
@ow0x You are correct.
[cython](https://pypi.org/project/Cython), [frozenlist](https://pypi.org/project/frozenlist), and [yarl](https://pypi.org/project/yarl) have all been updated and release since the original commit message.
The problem now seems to be [apipkg](https://pypi.org/project/apipkg) which is tested on Py3.11 but seems to fail here.

**Comment by cclauss** [comment]
Sorry.  It will take someone with steadier hands than mine to make this green on Py3.11.  I sense that `master` is not even the correct branch to work on.

**Comment by Dreamsorcerer** [comment]
We'll want master working and then the 3.9 branch too (and maybe 3.8 if it's not much extra work).

**Comment by Dreamsorcerer** [comment]
I think the apipkg error is caused by an outdated pytest (judging by the error reports https://github.com/pytest-dev/apipkg/issues/30, https://github.com/pytest-dev/py/issues/273, https://github.com/pytest-dev/pytest/issues/9181).

**Comment by webknjaz** [comment]
> I think the apipkg error is caused by an outdated pytest

Yes, confirmed by upgrading it.

**Pull request closed without merging**